### PR TITLE
Musashi  retrace spell and quest

### PR DESCRIPTION
**Pull request by Musashi-**

### CODE REVIEW

**Comment by TeoTwawki** [comment]
If you can fix your spaceship and rebase/collapse commits, that'd be appreciated. Some of your indents are off and you have a mix of tabs and spaces, and teh project wants 4 space indents

In most code editors you can configure them so that the tab key automatically creates 4 spaces, so you don't have to count them out..

```lua
function thing(argument1, argument2)
    if (conditions) then
        stuff inside conditional check
        if (conditions inside conditional check) then
            more stuff    
        end
    end
end;
```
Should look like that above. Below is ugly code.
```lua
function thing(argument1,argument2)
   if( conditions) then
		stuff inside conditional check
	   if(conditions inside conditional check )then
	   more stuff	 
	   end
 end
end;
```

**Comment by Musashi-** [comment]
ahh I see what your sayin there, thanks Teo appreciate it.

**Comment by Musashi-** [comment]
So is there anything else I have to do to this? Or has it been fixed now and merged into the code?

**Comment by teschnei** [comment]
What Teo said was what you have to fix still

**Comment by Musashi-** [comment]
Ok.. which files o.O my tab does = 4 spaces. Just checked, Press tab.. backspace, spacebar 4x, same thing. Where was the errors or what do I need to change in settings so its not so wierd for you guys.

**Comment by TeoTwawki** [comment]
Your files her eon github very obviously have actual tabs.

File: scripts/globals/spells/retrace.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+require("scripts/globals/teleports");
+-----------------------------------------
+-- OnSpellCast
+-----------------------------------------
+
+function onMagicCastingCheck(caster,target,spell)
+	local Allegiance = target:getCampaignAllegiance();
+	if (Allegiance == 1) then
+	return 0;
+	elseif (Allegiance == 2) then
```
Comment (TeoTwawki) [inline scripts/globals/spells/retrace.lua]: missing indent example

File: scripts/globals/teleports.lua [hunk 2]
```diff
@@ -424,7 +425,21 @@ function toMaw(player, option)
 			--Xarcabard (H-8)
 	end;
 end;
+-----------------------------------
```
Comment (TeoTwawki) [inline scripts/globals/teleports.lua]: Missing line between end and next header

File: scripts/zones/East_Ronfaure_[S]/Zone.lua [hunk 3]
```diff
@@ -15,6 +15,17 @@ require("scripts/globals/missions");
 -----------------------------------
 
 function onInitialize(zone)
+
+	local mobids = {17109357,
+					17138041,
```
Comment (TeoTwawki) [inline scripts/zones/East_Ronfaure_[S]/Zone.lua]: I'm not even sure what happened there

**Comment by TeoTwawki** [comment]
Hope that helped.

**Comment by TeoTwawki** [comment]
Gonna go out on a limb and guess copy paste caused most or all of this.

**Comment by Musashi-** [comment]
Yup just copy/pasted it. Next time ill check the spacing. I just did a edit to the effects/teleport.lua did that turn out right?

**Comment by TeoTwawki** [comment]
What I do is I have a saved macro that auto converts any tabs to spaces for me, and also trims any trailing space/tabs off. 

If you can use teamviewer and have a few minutes on irc, I can show you how to do that and also how to collapse commits

**Comment by Musashi-** [comment]
yea I can pop in no prob. All these specifics are making this a pain to share anything lol Especially when I don't know the pro specifics of coding, only what i've taught myself reading the code here. I learned how to code here, this is the only coding i've ever done since I setup my server.

**Comment by TeoTwawki** [comment]
Some past code (cough limbus cough) has gotten near impossible to maintain from the messy condition it was in, so this is less painful than having to do later after it pile sup 20 cars high and needs the jaws of life to dig out which piece needs changed someday.

File: scripts/zones/Windurst_Waters_[S]/npcs/Door_Rhinostery.lua [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+-----------------------------------
+
+function onTrigger(player,npc)
+	if (player:getQuestStatus(CRYSTAL_WAR, THE_LOST_BOOK)==QUEST_AVAILABLE and player:hasKeyItem(925)) then
+	player:startEvent(143);
+	player:addQuest(CRYSTAL_WAR, THE_LOST_BOOK); 
+	elseif player:hasKeyItem(971) then
+	player:startEvent(144);
+	elseif (player:getQuestStatus(CRYSTAL_WAR, THE_LOST_BOOK)==QUEST_ACCEPTED) then
+	end
```
Comment (xdemolish) [inline scripts/zones/Windurst_Waters_[S]/npcs/Door_Rhinostery.lua]: 4 space indent for each item within control block (applies to control blocks within control blocks too)

Comment (Musashi-) [inline scripts/zones/Windurst_Waters_[S]/npcs/Door_Rhinostery.lua]: Yea I think I got a good grasp on it yesturday, Teo and I did some
teamviewer / anydesk and went over quite a lot. In my opinion this is all a
lot more complicated than it has to be and very strict and picky(not you
guys, the process) but I do understand it helps with cleanliness of the
code, performance, as well as when its pushed through and auto scanned it
saves you guys a lot of legwork looking for errors / mistakes. Hopefully I
can actually share something with quality work soon enough. I still have
recall quests to share, dark ixion, and some rune fencer stuff
(abiliites/rune enhancement etc). Started assaults currently as well, just
want enough time to get some work done here before I have to get back to a
busy schedule again.

**Comment by teschnei** [comment]
the indentation of blocks still needs to be fixed, and the east_ronfaure zone is using tabs instead of spaces still

**Comment by Fiocitrine** [comment]
Is this gonna be fixed or just sit in PR limbo?

**Comment by Musashi-** [comment]
When I get the time to play with spaces, or somebody else can. Been busy
with r/l stuff

**Pull request closed without merging**